### PR TITLE
[SPARK-29788][DOC] Fix the typos in the SQL reference documents

### PR DESCRIPTION
**Pull request by jobitmathew**
### What changes were proposed in this pull request?

Fixing the typos in SQL reference document.

### Why are the changes needed?

For user readability

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Tested manually.

### CODE REVIEW

**Comment by jobitmathew84** [comment]
@gatorsmile,please review

File: docs/sql-ref-syntax-aux-show-table.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ SHOW TABLE EXTENDED [IN|FROM database_name] LIKE 'identifier_with_wildcards' [PA
     Specifies the regular expression pattern that is used to filter out unwanted tables.
     <ul> 
        <li> Except for `*` and `|` character, the pattern works like a regex.</li>
-       <li> `*` alone matches 0 or more characters and `|` is used to separate multiple different regexes,
+       <li> `*` alone matches 0 or more characters and `|` is used to separate multiple regex,
```
Comment (HyukjinKwon) [inline docs/sql-ref-syntax-aux-show-table.md]: Isn't `regexes` fine?

**Review by HyukjinKwon — APPROVED** [review]
looks good otherwise.

**Comment by HyukjinKwon** [comment]
ok to test

File: docs/sql-ref-syntax-aux-show-table.md (hunk 1 above)
Comment (jobitmathew84) [reply docs/sql-ref-syntax-aux-show-table.md]: @HyukjinKwon I think singular or plural it is regex,not regexes.

Comment (srowen) [reply docs/sql-ref-syntax-aux-show-table.md]: Hm, I'd say regexes

**Review by srowen — COMMENTED** [review]
Thanks, these are good. I might undo the regex change as it's not clear that's wrong.

File: docs/sql-ref-syntax-aux-show-table.md (hunk 1 above)
Comment (jobitmathew) [reply docs/sql-ref-syntax-aux-show-table.md]: O thanks

**Comment by jobitmathew** [comment]
@srowen and @HyukjinKwon changes done .Can you please check it.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**